docs/governance/rfc/005-python-implementation.md

```diff
@@ -0,0 +1,134 @@
+
+Further complicating matters is the Python community is divided regarding the right asynchronous implementation.
+[asyncio](https://docs.python.org/3/library/asyncio.html) is the standard library's answer, but you can find adherents of [trio](https://trio.readthedocs.io/en/stable/) and other asynchronous event loops.
+
+Python's immature support for http/2 is also a major challenge, particularly for servers.
+
+## Non-goals
+
+In order to make incremental progress, we explicitly cross off certain features in the initial pass at a Python implementation:
+
+1. *No cross-protocol support*: this will only be a client and server of the Connect protocol, not gRPC or grpc-web.
+2. *No full-duplex streams*: http/2 transport is hard to support on both client and server; half-duplex is fine though
+
+### Justification for only supporting Connect on the server
+
+gRPC is a poor match to Python's most commonly used asynchronous server framework, ASGI.
+A correct gRPC implementation requires control over very low-level details of the connection state and http/2 transport.
+These details are not exposed by ASGI, so supporting them would prevent connect-python's servers from being ASGI applications.
+
+That's an unacceptable trade-off to make because ASGI interoperability is important for integrating connect servers into larger applications, like those running on Django, Starlette or FastAPI.
+
+### Justification for only supporting Connect on the client
+
+Implementing a correct gRPC client again requires low-level control over the transport stack.
+The connect client would need to accept an extremely low-level and generic transport, even when contacting connect servers, just to support gRPC.
+
+This is more acceptable than the server case, but still unpleasant.
+Users will expect that they can pass an httpx.Client or similar to configure instrumentation, authentication, and so on.
+If we use a low-level library like hyper-h2 to implement the client, we'd likely need to build a bespoke abstraction supporting features like authentication which would be quite a bit more scope to cover.
+
+### Justification for only supporting half-duplex streams
+
+This is related to the above two justifications: http/2 is poorly supported in Python today.
+
+## Goals
+
+Fully pass conformance tests, with the following configuration:
+```
+features:
+  versions:
+    - HTTP_VERSION_1
+  protocols:
+    - PROTOCOL_CONNECT
+  codecs:
+    - CODEC_PROTO
+    - CODEC_JSON
+  compressions:
+    - COMPRESSION_IDENTITY
+    - COMPRESSION_GZIP
+    - COMPRESSION_BR
+    - COMPRESSION_ZSTD
+  stream_types:
+    - STREAM_TYPE_UNARY
+    - STREAM_TYPE_CLIENT_STREAM
+    - STREAM_TYPE_SERVER_STREAM
+    - STREAM_TYPE_HALF_DUPLEX_BIDI_STREAM
+  supports_half_duplex_bidi_over_http1: true
+  supports_h2c: false
+```
+
+## General design
+
+Most of the code for both servers and clients will be in a connect-python runtime library, installable with `pip install connect-python`.
```

Comment:
Why not `connectrpc`. The `python` suffix seems superfluous. We do the same for [dart](https://pub.dev/packages/connectrpc).